### ISSUE TITLE
Parent schema IDs don't appear to be properly populated in embedded schemas

### ISSUE THREAD

**Issue by brado** [michalmuskala/mongodb_ecto#63]
I have a very basic MongoDB schema structure, with model (document) "A" having an embeds_many relationship with model (document) "B". When I insert a new A containing several children B, looking at what is persisted in MongoDB, each B child has a field (which was not explicitly defined in its schema) which is supposed to hold the A parent's _id value. Instead, this parent ID value for all children is NULL. As this field is automatically added, it should properly reflect that parent's ID.

**Comment by brado** [comment]
I should mention that this behavior resulted from having a `belongs_to` in the embedded schema of an `embeds_many`. I am guessing that the `belongs_to` causes the parent ID to be present in the inserted embedded document, because without `belongs_to`, the parent ID doesn't appear. However, it never is populated.

**Comment by michalmuskala** [comment]
`belongs_to` should only be used with `has_many` or `has_one`. It's not useful to have `parent_id` in embedded document - it always belongs to the document it's embedded in. 

Because of this the current behaviour is to be expected. I agree there's room for improvement with the documentation though.

**Comment by brado** [comment]
@michalmuskala Ok....I'll buy this. I agree that for an embedded document, containment defines the relationship. I had pondered the possibility that queries on embedded documents might benefit (or be more performance optimal) if the ability existed to store that parent id. 

The interesting thing about this is that it was something I discovered on accident when trying to adapt that article by @josevalim on Ecto associations and embeds. It wasn't immediately clear whether `belongs_to` was needed (or required) or not. (Another related issue was that embedded documents apparently do not need to be defined using `embedded_schema do`, if the parent declares them in its `embeds_many`, it appears they can use the standard `schema "myobjects" do` syntax and embedding will work the same). 

So after I discovered this I began to ponder the potential uses for parent ID's in embedded docs. There may be some, but until I come across a specific use-case, closing this is cool, if the intent was that `belongs_to` explicitly not be used in embedded documents.
